shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -1293,24 +1294,30 @@ routing message to a random node so that:
                     body: ResponseBody::Fetch(fetch),
                     ..
                 })) => {
-                    for response in &fetch.responses {
-                        for partition in &response.partitions {
+                    // Clear this optional field to avoid making clients try to bypass shotover
+                    // partition.current_leader and partition.preferred_read_replica are cleared due to the same reason
+                    fetch.node_endpoints.clear();
+                    for fetch_response in &mut fetch.responses {
+                        for partition in &mut fetch_response.partitions {
+                            partition.current_leader = LeaderIdAndEpoch::default();
+                            partition.preferred_read_replica = BrokerId(-1);
```

Comment:
I checked to see if this field was used in non error cases.
The DB is sending protocol version 15 fetch responses and they all have BrokerId(-1) so I think its safe to assume this is only used in error cases.

So, looks good :+1: